felt/src/arbitrary.rs

```diff
@@ -0,0 +1,39 @@
+use num_bigint::BigUint;
+use num_traits::Zero;
+use proptest::prelude::*;
+
+use crate::{
+    bigint_felt::{FeltBigInt, FIELD_HIGH, FIELD_LOW},
+    Felt252,
+};
+
+pub(crate) fn any_felt_big_int<const PH: u128, const PL: u128>(
+) -> impl Strategy<Value = FeltBigInt<PH, PL>> {
+    (0..=PH)
+        // turn range into `impl Strategy`
+        .prop_map(|x| x)
+        // choose second 128-bit limb capped by first one
+        .prop_flat_map(|high| {
+            let low = if high == PH {
+                Strategy::sboxed((0..PH).prop_map(|x| x))
+            } else {
+                Strategy::sboxed(any::<u128>())
+            };
+            (Just(high), low)
+        })
+        // turn (u128, u128) into BigUint and then into FeltBigInt
+        .prop_map(|(high, low)| {
+            let biguint = (BigUint::from(high) << 128) + low;
+            FeltBigInt::from(biguint)
+        })
+}
+
+/// Returns a [`Strategy`] that generates any valid Felt252
+pub fn any_felt252() -> impl Strategy<Value = Felt252> {
+    any_felt_big_int::<FIELD_HIGH, FIELD_LOW>().prop_map(|value| Felt252 { value })
+}
```

Comment:
I think these come for free as `any::<T>` as long as the types implement the `Arbitrary` trait.

Reply:
Yeah, it's better to impl the trait. I started doing it and quickly got stuck because in the trait you have to be explicit with the return type of its method (i.e. no `impl Strategy` magic), so it requires a bit more thought. Now that I have this implementation in place, I think that issue could be solved just by using a boxed strategy. I'll check this later.

Reply:
Done. Although I haven't found an easy way to generate a nonzero felt with the trait. As far as I know there are two ways:
- use `prop_filter`, like the function `nonzero_felt252` does
- receive some parameters that change the generation strategy in the trait, and use `any_with::<Felt252>(args)` instead of plain `any::<Felt252>()`